include/ginkgo/core/base/exception.hpp

```diff
@@ -269,6 +269,33 @@ class DimensionMismatch : public Error {
 };
 
 
+/**
+ * BadDimension is thrown if an operation is being applied to a LinOp
+ * with bad dimensions.
+ */
+class BadDimension : public Error {
+public:
+    /**
+     * Initializes a bad dimension error.
```

Comment:
I am not sure whether we also have the format for the comments.
There is one line between the description of the function and the parameter in other files.